src/SellingPartnerApiServiceProvider.php

```diff
@@ -56,6 +50,10 @@ public function boot(): void
      */
     public function register(): void
     {
+        if (!config('spapi.registration_enabled', true)) {
+            return;
+        }
```

Comment:
Well, not essentially to make mocking easier, but rather to create a setup for testing that ensures that there not any outgoing requests made with real credentials. I now see it more as a security switch. On top of this the developer can easily create mocked objects and can be sure that there only mocked requests made. But actually the value of this switch decreased, after I also the binding from singleton to normal. Before, the singleton instances would directly be created and each of them would ensure there is a valid auth token. So there was basically no way to hook in mocked objects, because this was all done automatically before the test method was actually called.

In other words: With singleton binding 'auto-registration' also meant potential accidental api requests before each test although the tested code might even have nothing to do with spapi at all. 
With normal binding the purpose of this switch is reduced, because these requests are only made when the api classes are actually used. This way it is enough to replace/mock the classes in the tests to avoid accidental requests.

But since there is no built-in mocking and mocking these libraries is very rough to do, I think a hard switch like this still provides some value ;)